Formula/p/php.rb

```diff
@@ -209,37 +209,44 @@ def install
       --with-zlib
     ]
 
-    if OS.mac?
+    intl_arg = if OS.mac?
       args << "--enable-dtrace"
       args << "--with-ldap-sasl"
-      args << "--with-os-sdkpath=#{MacOS.sdk_path_if_needed}"
+      "--disable-intl"
     else
       args << "--disable-dtrace"
       args << "--without-ldap-sasl"
       args << "--without-ndbm"
       args << "--without-gdbm"
+      "--enable-intl"
     end
 
-    system "./configure", *args
+    system "./configure", *args, *intl_arg
     system "make"
     system "make", "install"
 
+    if OS.mac?
+      # Build C++ extension modules with Clang
+      system "./configure", *args, "--disable-opcache", "--enable-intl=shared"
+      ENV.method(DevelopmentTools.default_compiler).call
+      system "make", "install-modules"
```

Comment:
Seems kinda suspicious to configure with one compiler but build with another.

Reply:
It could be built in separate formula but would need to make sure it doesn't cause problems we've hit before from split formulae (e.g. old GStreamer).

Reply:
Php searches the location of EXTENSION_DIR env variable when php was built, so there's no need to add rpath.

Reply:
The build-time `EXTENSION_DIR` may not be the same as the run-time `EXTENSION_DIR`, though.

Reply:
The EXTENSION_DIR environment variable has no effect at runtime. It can only be overwritten in php.ini, at which point it would be expected not to load an extension outside that directory. 

E.g.

`EXTENSION_DIR=/usr/lib/php/modules make install-modules`
php -d `"extension_dir=/home/username/php/modules" -d "extension=intl" --ri intl`

Should fail when there's no extension in /home/username/php/modules. With rpath to /usr/lib/php/modules it would unexpectedly (?) succeed.

Reply:
Actually, I'm not sure if rpath would even have an effect there. Let me test that.

Reply:
```console
[m@M static-php-cli]$ patchelf --print-rpath buildroot/bin/php
$ORIGIN/../modules
[m@M static-php-cli]$ buildroot/bin/php -d extension=intl --ri intl

Warning: PHP Startup: Unable to load dynamic library 'intl' (tried: /lib/php/extensions/no-debug-zts-20240924/intl (/lib/php/extensions/no-debug-zts-20240924/intl: cannot open shared object file: No such file or directory), /lib/php/extensions/no-debug-zts-20240924/intl.so (/lib/php/extensions/no-debug-zts-20240924/intl.so: cannot open shared object file: No such file or directory)) in Unknown on line 0
Extension 'intl' not present.
[m@M static-php-cli]$ ls buildroot/modules/intl.so
buildroot/modules/intl.so
```

Setting a rpath has no effect on php's extension loading mechanism.

Reply:
I am talking about `dyld`'s search mechanism for `dlopen`, which does use the binary's `LC_RPATH` commands when searching. The results on Linux aren't really relevant here.

And re `EXTENSION_DIR`: my point was that the `HOMEBREW_PREFIX` at build-time is not always the same as the `HOMEBREW_PREFIX` at runtime (because users can technically install Homebrew wherever they want). But it looks like the `php` bottles aren't relocatable, so that's probably moot (unless a user insists on using `--force-bottle`).